x-pack/platform/plugins/shared/stack_connectors/server/connector_types/webhook/get_axios_config.ts

```diff
@@ -0,0 +1,158 @@
+  services: { connectorTokenClient },
+  logger,
+  configurationUtilities,
+}: GetOAuth2AxiosConfigParams) => {
+  const { accessTokenUrl, clientId, scope, additionalFields, headers } = config;
+  const { clientSecret } = secrets;
+
+  // `additionalFields` should be parseable, we do check API schema validation and in
+  // action config validation step.
+  let parsedAdditionalFields;
+  try {
+    parsedAdditionalFields = additionalFields ? JSON.parse(additionalFields) : undefined;
+  } catch (error) {
+    logger.error(`Connector ${connectorId}: error parsing additional fields`);
+  }
+
+  let accessToken;
+  try {
+    accessToken = await getOAuthClientCredentialsAccessToken({
+      connectorId,
+      logger,
+      configurationUtilities,
+      oAuthScope: scope,
+      credentials: {
+        secrets: { clientSecret: clientSecret! },
+        config: {
+          clientId: clientId!,
+          ...(parsedAdditionalFields ? { additionalFields: parsedAdditionalFields } : {}),
+        },
+      },
+      tokenUrl: accessTokenUrl!,
+      connectorTokenClient,
+    });
+  } catch (error) {
+    throw new Error(`Unable to retrieve/refresh the access token: ${error.message}`);
+  }
+
+  if (!accessToken) {
+    throw new Error(`Unable to retrieve new access token`);
+  }
+  logger.debug(`Successfully retrieved access token`);
+
+  const { onFulfilled, onRejected } = getOauth2DeleteTokenAxiosInterceptor({
+    connectorTokenClient,
+    connectorId,
+  });
+  const axiosInstance = axios.create();
+  axiosInstance.interceptors.response.use(onFulfilled, onRejected);
+
+  const headersWithAuth = {
+    ...headers,
+    'Content-Type': 'application/json',
+    Authorization: accessToken,
+  };
+
+  return { axiosInstance, headers: headersWithAuth, sslOverrides: {} };
+};
+
+interface GetDefaultAxiosConfig {
+  config: ConnectorTypeConfigType;
+  secrets: ConnectorTypeSecretsType;
+  headers: Record<string, string> | null;
+}
+const getDefaultAxiosConfig = async ({ config, secrets, headers }: GetDefaultAxiosConfig) => {
+  const { hasAuth, authType, verificationMode, ca } = config;
+
+  const axiosInstance = axios.create();
+  const { basicAuth, sslOverrides } = buildConnectorAuth({
+    hasAuth,
+    authType,
+    secrets,
+    verificationMode,
+    ca,
+  });
+
+  const headersWithAuth = combineHeadersWithBasicAuthHeader({
+    username: basicAuth.auth?.username,
+    password: basicAuth.auth?.password,
+    headers,
+  });
+
+  return { axiosInstance, headers: headersWithAuth, sslOverrides };
+};
+
+export interface GetAxiosConfigResponse {
+  axiosInstance: AxiosInstance;
+  headers: Record<string, AxiosHeaderValue> | undefined;
+  sslOverrides: SSLSettings;
+}
+
+export interface GetAxiosConfigParams {
+  config: ConnectorTypeConfigType;
+  secrets: ConnectorTypeSecretsType;
+  connectorId: string;
+  logger: Logger;
+  services: Services;
+  configurationUtilities: ActionsConfigurationUtilities;
+}
+export const getAxiosConfig = async ({
```

Comment:
this will be called from the webhook.executor function. In getDefaultAxiosConfig is the code that ran before. To not pollute the executor function even more, I created this function with the new behaviour (oauth2 token validation/refresh) inside of this condition

Reply:
Wouldn't this file be better in `common/auth`? This looks like OAuth2 specific code.